src/firefly/java/edu/caltech/ipac/firefly/server/query/ObsCoreMetadataQuery.java

```diff
@@ -0,0 +1,107 @@
+
+import edu.caltech.ipac.firefly.data.TableServerRequest;
+import edu.caltech.ipac.firefly.server.util.Logger;
+import edu.caltech.ipac.firefly.server.util.QueryUtil;
+import edu.caltech.ipac.table.DataGroup;
+import edu.caltech.ipac.table.DataObject;
+import edu.caltech.ipac.table.DataType;
+import edu.caltech.ipac.util.StringUtils;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.*;
+import java.util.stream.Collectors;
+
+import static edu.caltech.ipac.firefly.server.query.ObsCoreMetadataQuery.*;
+import static edu.caltech.ipac.firefly.server.query.UwsJobProcessor.getTableResult;
+
+
+@SearchProcessorImpl(id = ID, params = {
+        @ParamDoc(name = SVC_URL, desc = "base TAP url endpoint excluding '/sync'"),
+        @ParamDoc(name = OBSCORE_TNAME, desc = "obscore schema.table"),
+        @ParamDoc(name = COLUMNS, desc = "columns to retrieve metadata for"),
+})
+public class ObsCoreMetadataQuery extends EmbeddedDbProcessor {
+    public static final String ID = "ObsCoreMetadataQuery";
+    public static final String SVC_URL = "serviceUrl";
+    public static final String OBSCORE_TNAME = "obscoreTable";
+    public static final String COLUMNS = "columns";
+    public static final String QUERY_FRAGMENT = "/sync?REQUEST=doQuery&LANG=ADQL&QUERY=";
+    public static final Logger.LoggerImpl LOGGER = Logger.getLogger();
+
+
+    public DataGroup fetchDataGroup(TableServerRequest request) throws DataAccessException {
+        List<String> columns = StringUtils.asList(request.getParam(COLUMNS), ",");
+        List<Future<DataGroup>> futures = new ArrayList<>();
+        List<Throwable> exceptions = new ArrayList<>();
+
+        // Submit queries for each column in parallel
+        ExecutorService executor = Executors.newCachedThreadPool();
+        long sTime = System.currentTimeMillis();
+        for (String column : columns) {
+            Future<DataGroup> future = executor.submit(() -> {
+                long cur = System.currentTimeMillis();
+                DataGroup columnMetaData = fetchColumnMetaData(request, column);
+                LOGGER.debug(String.format("%s query returned in %d msec%n", column, System.currentTimeMillis() - cur));
+                return columnMetaData;
+            });
+            futures.add(future);
+        }
+
+        // Retrieve DataGroup from each future one by one
+        List<DataGroup> columnsMetadata = new ArrayList<>();
+        for (Future<DataGroup> future : futures) {
+            try {
+                columnsMetadata.add(future.get());  // note: blocking main thread until completion of this future, will still lead to completion of the futures running in other threads
+            } catch (InterruptedException | ExecutionException e) {
+                exceptions.add(e);
+                columnsMetadata.add(null);
+            }
+        }
+
+        LOGGER.debug(String.format("All queries returned in  %d msec%n", System.currentTimeMillis() - sTime));
```

Comment:
We have a convenient class(`StopWatch`) that will do this for you.  It is also hooked to logger, but at `TRACE` level.  This allow you to set as many debugging logs as you want during development and not have it show up in normal runtime logs.